### ISSUE TITLE
BuildConfig java is not compiled (only generated) when 'compile' task called

### ISSUE THREAD

**Issue by fboldog** [cbeust/kobalt#51]
To reproduce:
add a buildConfig block into java or (kotlin)Project and call compile task

Tested with build 0.292 and examples/variants and examples/android-flavors

```kotlin
val p = kotlinProject {

    name = "variants"
    group = "com.beust.kobalt.example"
    artifactId = name
    version = "0.1"

    buildConfig {
        field("freeField", "String", "\"The free field\"")
    }
....
```

**Comment by cbeust** [comment]
Ok github is being a bit too aggressive in parsing the commit message, it's a tentative fix, waiting for @fboldog to confirm.

**Comment by fboldog** [comment]
Still valid with 0.301.

**Comment by cbeust** [comment]
It's building fine for me on 0.301:

```
javac -d ./kobaltBuild/intermediates/classes/pro/debug -classpath ./kobaltBuild/generated/classes:/Users/beust/android/adt-bundle-mac-x86_64-20140702/sdk/platforms/android-23/android.jar -source 1.6 -target 1.6 -nowarn ././kobaltBuild/generated/source/com/beust/kobalt/android/helloworld/BuildConfig.java ./src/main/java/com/beust/kobalt/android/helloworld/HelloWorldActivity.java
```

How are you reproducing?

**Comment by cbeust** [comment]
Fix deployed in 0.302.
